lib/playground.dart

```diff
@@ -315,21 +341,47 @@ class Playground {
     if (context.focusedEditor == 'dart') {
       ga.sendEvent('main', 'help');
 
+      var input;
       Position pos = editor.document.cursor;
-      var input = new SourceRequest()
+      int offset = editor.document.indexFromPos(pos);
+
+      if (querySelector(".CodeMirror-hint-active") != null) {
+        String completionText = querySelector(".CodeMirror-hint-active").text;
+        var source = context.dartSource;
+        int lastSpace = source.substring(0, offset).lastIndexOf(" ") + 1;
+        int lastDot = source.substring(0, offset).lastIndexOf(".") + 1;
+        offset = max(lastSpace, lastDot);
+        source = _context.dartSource.substring(0, offset)
```

Comment:
You're creating some synthetic source based on if the current completion was chosen, and asking for completion info for that source? Very inventive! Can you add a line comment to that effect, to make it easier to parse this code?